### PR TITLE
Fix and add many tests

### PR DESCRIPTION
**Pull request by harshil21**
PR changes:
- Adds and fixes all tests
- Fixes all mock sims
- Adds integration test, which runs on interest launch data and winter_2023_launch data.
- Check code coverage: 86%. imu and mock_imu.py are mostly the ones not being covered.

Not doing: 

- Add bash script to auto install mscl depending on platform
- Add mscl installation to CI
- Add windows runner to the CI?
- Check if we can mock the mscl library in test_imu to fully test `_fetch_data_loop`

### CODE REVIEW

**Comment by JacksonElia** [comment]
TODO looks good except for: I don't think we need to auto install mscl--it's a lot of work, plus the imu won't even work for our laptops locally, plus I don't think the imu class will be changed that much in the future.

**Comment by harshil21** [comment]
> TODO looks good except for: I don't think we need to auto install mscl--it's a lot of work, plus the imu won't even work for our laptops locally, plus I don't think the imu class will be changed that much in the future.

Okay. It's just that from a python packaging perspective, it makes sense to install the dependencies from one place. Tests for this will probably be...long.

File: airbrakes/utils.py [hunk 1]
```diff
@@ -10,23 +10,28 @@
 MOVE_CURSOR_UP = "\033[F"  # Move the cursor one line up
 
 
-def convert_to_nanoseconds(value) -> int:
-    """Converts seconds to nanoseconds, if `value` is in float."""
+def convert_to_nanoseconds(timestamp_str: str) -> int | None:
```
Comment (JacksonElia) [inline airbrakes/utils.py:13]: maybe we could do a more sophisticated check? e.g. check if the number passed into here is more than 10^9?

Comment (harshil21) [reply airbrakes/utils.py:13]: yeah perhaps. But the better solution is to just clean the csv data in the first place

**Comment by JacksonElia** [comment]
LGTM so far. Can we try to merge this soon? Also, can we merge #37 into this?

**Comment by harshil21** [comment]
@JacksonElia Yeah, I'm just waiting for you to write tests in #37 so we can merge that into this and then merge this.

**Review by JacksonElia — APPROVED** [review]
Lgtm

**Pull request merged**